File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Response/Filter/CoreFilterAssignElementMasterDTO.js
import CoreFilterAssignAbstract from "./CoreFilterAssignAbstract.js";
import ConvertUtil from "../../../Common/ConvertUtil.js";
import CoreButtonAssignElementDTO from "../Button/CoreButtonAssignElementDTO.js";
import CoreTranslateDTO from "../Translate/CoreTranslateDTO.js";

export default class CoreFilterAssignElementMasterDTO extends CoreFilterAssignAbstract {
    constructor() {
        super();
    }

    getRecordId() {
        return this.recordId;
    }

    getCoreFilterAssignAbstractMap() {
        if (this.coreFilterAssignAbstractMap && !(this.coreFilterAssignAbstractMap instanceof Map)) {
            this.coreFilterAssignAbstractMap = ConvertUtil.ConvertGeneralWithMap(CoreFilterAssignAbstract, this.coreFilterAssignAbstractMap);
        }
        return this.coreFilterAssignAbstractMap;
    }

    getCoreButtonAssignElementDTOMap() {
        if (this.coreButtonAssignElementDTOMap && !(this.coreButtonAssignElementDTOMap instanceof Map)) {
            this.coreButtonAssignElementDTOMap = ConvertUtil.ConvertGeneralWithMap(CoreButtonAssignElementDTO, this.coreButtonAssignElementDTOMap, "id");
        }
        return this.coreButtonAssignElementDTOMap;
    }

    getCoreTranslateDTOMap() {
        if (this.coreTranslateDTOMap && !(this.coreTranslateDTOMap instanceof Map)) {
            this.coreTranslateDTOMap = ConvertUtil.ConvertGeneralWithMap(CoreTranslateDTO, this.coreTranslateDTOMap);
        }
        return this.coreTranslateDTOMap;
    }

    getUuid() {
        return this.uuid;
    }
}

File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/model/dto/response/filter/CoreFilterAssignElementMasterDTO.java
package org.infra.reactive.form.engine.form.engine.model.dto.response.filter;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.infra.reactive.form.engine.form.engine.model.dto.response.button.CoreButtonAssignElementDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.translate.CoreTranslateDTO;

import java.util.Map;

@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
public class CoreFilterAssignElementMasterDTO extends CoreFilterAssignAbstract {
    private long recordId;
    private String uuid;
    private Map<Long, CoreFilterAssignAbstract> coreFilterAssignAbstractMap;
    private Map<Long, CoreButtonAssignElementDTO> coreButtonAssignElementDTOMap;
    private Map<String, CoreTranslateDTO> coreTranslateDTOMap;
}
